app/src/main/java/org/gdg/frisbee/android/widget/FeedbackFragment.java

```diff
@@ -105,24 +116,66 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 .setTitle(R.string.feedback)
                 .setCancelable(true)
                 .setNegativeButton(R.string.feedback_cancel, null)
-                .setPositiveButton(R.string.feedback_send, new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
+                .setPositiveButton(R.string.feedback_send, null);
+
+        return builder.create();
+    }
+
+    public void onStart() {
+        super.onStart();
+        final AlertDialog dialog = (AlertDialog) getDialog();
+        if (dialog != null) {
+
+            Button positiveButton = dialog.getButton(Dialog.BUTTON_POSITIVE);
+            positiveButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+
+                    if (isValidInput()) {
                         mApi.setLoadingMessage(getActivity().getString(R.string.feedback_sending));
                         mApi.setCallback(new RestCallback() {
                             public void success(Object obj) {
                                 //TODO add feedback
-//                                Toast.makeText(getActivity(), obj.toString(), Toast.LENGTH_SHORT).show();
                                 mMessageField.setText("");
                                 mProperties = new JSONObject();
                             }
                         });
                         mApi.sendFeedback(mMessageField.getText().toString(),
                                 mEmailField.getText().toString(), mProperties, "");
+
+                        dialog.dismiss();
                     }
-                });
+                }
+            });
+        }
+    }
 
-        return builder.create();
+    private boolean isValidInput() {
+
+        boolean isValid;
+
+        String strMessage = mMessageField.getText().toString();
+        String strEmail = mEmailField.getText().toString();
+
+        if (!TextUtils.isEmpty(strMessage)) {
+            isValid = true;
+            mLayoutMessage.setErrorEnabled(false);
+        } else {
+            isValid = false;
+            mLayoutMessage.setErrorEnabled(true);
+            mLayoutMessage.setError(getString(R.string.feedback_message_required));
+        }
+
+        if (!TextUtils.isEmpty(strEmail) && android.util.Patterns.EMAIL_ADDRESS.matcher(strEmail).matches()) {
+            isValid = true;
```

Comment:
Yes it will help and we can also take two different variables if memory is
not the concern but I really wanted to have a single variable to accomplish
this task!

On Mon, Jul 27, 2015, 7:49 PM Said Tahsin Dane <notifications@github.com>
wrote:

> In app/src/main/java/org/gdg/frisbee/android/widget/FeedbackFragment.java
> <https://github.com/gdg-x/frisbee/pull/412#discussion_r35540194>:
>
> > +        boolean isValid;
> > +
> > +        String strMessage = mMessageField.getText().toString();
> > +        String strEmail = mEmailField.getText().
>
> toString();
> > +
> > +        if (!TextUtils.isEmpty(strMessage)) {
> > +            isValid = true;
> > +            mLayoutMessage.setErrorEnabled(false);
> > +        } else {
> > +            isValid = false;
> > +            mLayoutMessage.setErrorEnabled(true);
> > +            mLayoutMessage.setError(getString(R.string.feedback_message_required));
> > +        }
> > +
> > +        if (!TextUtils.isEmpty(strEmail) && android.util.Patterns.EMAIL_ADDRESS.matcher(strEmail).matches()) {
> > +            isValid = true;
>
> I think isValid = isValid && true; will help.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gdg-x/frisbee/pull/412/files#r35540194>.
>